src/frontend/components/OptionField/OptionsField.jsx

```diff
@@ -5,39 +5,107 @@ import {
   MdPlaylistAdd,
   AiOutlineLike,
   BsThreeDotsVertical,
+  AiOutlineDislike,
 } from "../../constants/react-icons";
 import { useOnClickOutside } from "../../hooks/onClickOutside";
 import { PlayListModal } from "../PlayListModal/PlayListModal";
+import { getService } from "../../services/combinedService";
+import { useAuth } from "../../context/auth-context";
+import { useWatchlater } from "../../context/watchlater-context";
+import { useLike } from "../../context/like-context";
+import { ADD_TO_LIKE, RMV_FROM_LIKE } from "../../constants/like-constants";
+import {
+  ADD_TO_WATCHLATER,
+  RMV_FROM_WATCHLATER,
+} from "../../constants/watchlater-constants";
 
-export const OptionsField = ({vid}) => {
+export const OptionsField = ({ vid }) => {
   const [showOptions, setShowOptions] = useState(false);
-    const [showModal,setShowModal] = useState(false);
+  const [showModal, setShowModal] = useState(false);
   const closeModal = () => setShowModal(false);
   const openModal = () => setShowModal(true);
   const optionsRef = useRef();
   useOnClickOutside(optionsRef, () => setShowOptions(false));
+
+  const { watchlaterState, watchlaterDipatcher } = useWatchlater();
+  const { likeState, likeDipatcher } = useLike();
+
+  const {
+    user: { tokenVal },
+  } = useAuth();
+  const AddToLike = async () => {
+    const data = await getService("likes", "post", tokenVal, vid);
```

Comment:
Superb implementation of `getService`! 👌

Reply:
Thanks